clang fails with a cast "loses const" error

<!--- WARNING: This is an issue tracker. Before opening a new issue make sure you read https://github.com/PointCloudLibrary/pcl/blob/master/CONTRIBUTING.md#using-the-issue-tracker. -->

<!--- Provide a general summary of the issue in the Title above -->
Trunk does not compile with clang , besause of a reinterpret cast loses the const , on bivariate_polinomial.hpp. 

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Operating System and version: Ubuntu 18.04
* Compiler: clang-10
* PCL Version: trunk

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->

<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
I compiles with clang

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->
Fails at 
`pcl/pcl/common/include/pcl/common/impl/bivariate_polynomial.hpp:269:13: error: reinterpret_cast from 'const int *' to 'char *' casts away qualifiers
  os.write (reinterpret_cast<char*> (&degree), sizeof (int));`


## Code to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
trunk , using clang-10

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
A Quick fix is to just add the const to bivariate_polynomial.hpp:269 and bivariate_polynomial.hpp:271
<!--- or ideas how to implement the addition or change -->

--- a/common/include/pcl/common/impl/bivariate_polynomial.hpp
+++ b/common/include/pcl/common/impl/bivariate_polynomial.hpp
@@ -266,9 +266,9 @@ pcl::operator<< (std::ostream& os, const pcl::BivariatePolynomialT<real>& p)
 template<typename real> void
 pcl::BivariatePolynomialT<real>::writeBinary (std::ostream& os) const
 {
-  os.write (reinterpret_cast<char*> (&degree), sizeof (int));
+  os.write (reinterpret_cast<const char*> (&degree), sizeof (int));
   unsigned int paramCnt = getNoOfParametersFromDegree (this->degree);
-  os.write (reinterpret_cast<char*> (this->parameters), paramCnt * sizeof (real));
+  os.write (reinterpret_cast<const char*> (this->parameters), paramCnt * sizeof (real));
 }
 
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

